scio-core/src/main/scala/com/spotify/scio/values/SideInput.scala

```diff
@@ -48,7 +48,9 @@ trait SideInput[T] extends Serializable {
   }
 
   /**
-   * Create a new [[SideInput]] by applying a wrapping this SideInput with a mapper.
+   * Create a new [[SideInput]] by applying a function on the elements wrapped in this SideInput.
+   *
+   * The function is applied lazily at the time of reading the side input.
```

Comment:
I am wondering if this can be misused as the fn is applied at the time of reading from the `SideInput`. It might be a nice to have a way to apply the functions before the `SideInput` is constructed.

Reply:
The result of `map` is cached so it's not a problem (see `SideInput#getCache`). I'd remove this line to avoid confusion.